### ISSUE TITLE
Fields of type"typing.Annotated" are not supported in pydantic

### ISSUE THREAD

**Issue by maarten0912** [gabrieldemarmiesse/python-on-whales#490]
When importing `python_on_whales` with `pydantic` version 1.7.4, I get the following error:

```
    from python_on_whales import docker
  File "/lib/python3.10/site-packages/python_on_whales/__init__.py", line 2, in <module>
    from .components.buildx.cli_wrapper import Builder
  File "/lib/python3.10/site-packages/python_on_whales/components/buildx/cli_wrapper.py", line 9, in <module>
    import python_on_whales.components.image.cli_wrapper
  File "/lib/python3.10/site-packages/python_on_whales/components/image/cli_wrapper.py", line 16, in <module>
    from python_on_whales.components.container.cli_wrapper import (
  File "/lib/python3.10/site-packages/python_on_whales/components/container/cli_wrapper.py", line 30, in <module>
    from python_on_whales.components.container.models import (
  File "/lib/python3.10/site-packages/python_on_whales/components/container/models.py", line 55, in <module>
    class ContainerDeviceRequest(DockerCamelModel):
  File "/lib/python3.10/site-packages/pydantic/main.py", line 262, in __new__
    fields[ann_name] = inferred = ModelField.infer(
  File "/lib/python3.10/site-packages/pydantic/fields.py", line 315, in infer
    return cls(
  File "/lib/python3.10/site-packages/pydantic/fields.py", line 284, in __init__
    self.prepare()
  File "/lib/python3.10/site-packages/pydantic/fields.py", line 356, in prepare
    self._type_analysis()
  File "/lib/python3.10/site-packages/pydantic/fields.py", line 510, in _type_analysis
    raise TypeError(f'Fields of type "{origin}" are not supported.')
TypeError: Fields of type "<class 'typing.Annotated'>" are not supported.
```

Maybe a newer version of pydantic is required, while `requirements.txt` only requires version 1.5 or higher, except 2.0.x.

Reference for [Annotated](https://docs.pydantic.dev/latest/concepts/types/#custom-types)

**Comment by gabrieldemarmiesse** [comment]
Thank you for the bug report! Indeed it seems that the version range is wrong for pydantic. We should find which version of pydantic allows Annotated and us that version as the lower bound

**Comment by gabrieldemarmiesse** [comment]
This is fixed by @kashyab12  in #491 , I'll make a new release.
